### PR TITLE
Updated to restore dependencies recursively

### PR DESCRIPTION
**Pull request by josephspurrier**
The issue I see is if you do a gvt restore, it will download the dependencies (ball and beachball), but it won't run a gvt restore on beachball which also has a vendor and manifest with dependencies.

I created a few test packages so you can see how I'm testing: https://github.com/josephspurrier/vendorball

My code is messy, but it's a start. I did not test extensively.

### CODE REVIEW

File: restore.go [hunk 1]
```diff
@@ -119,5 +124,22 @@ func downloadDependency(dep vendor.Dependency) error {
...
 
+	// Check for for manifests in dependencies
+	man := filepath.Join(dst, "vendor", "manifest")
+	venDir := filepath.Join(dst, "vendor")
+	if _, err := os.Stat(man); err == nil {
+		m, err := vendor.ReadManifest(man)
+		if err != nil {
+			return fmt.Errorf("could not load manifest: %v", err)
+		}
+		var errors uint32
```
Comment (FiloSottile) [inline restore.go:135]: This variable is never looked at, so the error would not be reported

**Comment by FiloSottile** [comment]
Thanks, this makes sense! I'll merge and then add error handling and loop detection.

File: restore.go [hunk 2]
```diff
@@ -119,5 +124,22 @@ func downloadDependency(dep vendor.Dependency) error {
 		return err
 	}
 
+	// Check for for manifests in dependencies
+	man := filepath.Join(dst, "vendor", "manifest")
+	venDir := filepath.Join(dst, "vendor")
+	if _, err := os.Stat(man); err == nil {
+		m, err := vendor.ReadManifest(man)
+		if err != nil {
+			return fmt.Errorf("could not load manifest: %v", err)
```
Comment (FiloSottile) [inline restore.go:133]: A failure to read a manifest should not be a fatal error, it might not be a gvt manifest

**Comment by josephspurrier** [comment]
Awesome, great work! I'll definitely be adding this to all my projects.

**Comment by FiloSottile** [comment]
Actually, loops can't happen, so no need to detect them.

But a vendor folder can be anywhere, so will have to scan the whole tree.

**Comment by josephspurrier** [comment]
Ah, I guess a project could vendor another package that has just a manifest in their vendor folder, I didn't think about that.

**Pull request merged**